diff --git a/motion.h b/motion.h
index c08d84f..f6b9d58 100644
--- a/motion.h
+++ b/motion.h
@@ -52,7 +52,7 @@
 
 #define _LINUX_TIME_H 1
 #if defined(HAVE_LINUX_VIDEODEV_H) && (!defined(WITHOUT_V4L)) && (!defined(BSD))
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #endif
 
 #include <pthread.h>
diff --git a/track.c b/track.c
index 91e89d7..36e2600 100644
--- a/track.c
+++ b/track.c
@@ -10,6 +10,7 @@
 #include "motion.h"
 
 #if defined(HAVE_LINUX_VIDEODEV_H) && (!defined(WITHOUT_V4L))
+#include <linux/videodev2.h>
 #include "pwc-ioctl.h"
 #endif
 
diff --git a/video.h b/video.h
index 12f2928..5451b64 100644
--- a/video.h
+++ b/video.h
@@ -15,7 +15,7 @@
 
 
 #if defined(HAVE_LINUX_VIDEODEV_H) && (!defined(WITHOUT_V4L))
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include "vloopback_motion.h"
 #include "pwc-ioctl.h"
 #endif
